dipy/stats/analysis.py

```diff
@@ -147,27 +114,73 @@ def dti_measures(bundle, metric, dt, pname, bname, subject, group, ind, dir):
             path of output directory
     """
 
-    dt["bundle"] = []
-    dt["disk#"] = []
+    if group == 'patient':
+        group_id = 1  # 1 means patient
+    else:
+        group_id = 0  # 0 means control
+
+    dt["streamline"] = []
+    dt["disk"] = []
     dt["subject"] = []
     dt[pname] = []
     dt["group"] = []
 
     values = map_coordinates(metric, bundle._data.T,
                              order=1)
 
-    dt["disk#"].extend(ind[list(range(len(values)))]+1)
-    dt["bundle"].extend([bname]*len(values))
+    dt["disk"].extend(ind[list(range(len(values)))]+1)
     dt["subject"].extend([subject]*len(values))
-    dt["group"].extend([group]*len(values))
+    dt["group"].extend([group_id]*len(values))
     dt[pname].extend(values)
 
-    _save_hdf5(os.path.join(dir, pname), dt, col_name="bundle")
+    for st_i in range(len(bundle)):
+
+        st = bundle[st_i]
+        dt["streamline"].extend([st_i]*len(st))
+
+    file_name = bname+"_"+pname
+
+    _save_hdf5(os.path.join(dir, file_name), dt, col_name="bundle")
+
+
+def assignment_map(target_bundle, model_bundle, no_disks):
+    """
+    Calculates assignment maps of the traget bundle with reference to
+    model bundle centroids.
+
+    Parameters
+    ----------
+    target_bundle : streamlines
+    model_bundle : streamlines
+    no_disks : integer, optional
+        Number of disks used for dividing bundle into disks. (Default 100)
+
+    References
+    ----------
+    .. [Chandio19] Chandio, B.Q., S. Koudoro, D. Reagan, J. Harezlak,
+    E. Garyfallidis, Bundle Analytics: a computational and statistical
+    analyses framework for tractometric studies, Proceedings of:
+    International Society of Magnetic Resonance in Medicine (ISMRM),
+    Montreal, Canada, 2019.
+    """
+
+    mbundle_streamlines = set_number_of_points(model_bundle,
+                                               nb_points=no_disks)
 
+    metric = AveragePointwiseEuclideanMetric()
+    qb = QuickBundles(threshold=85., metric=metric)
+    clusters = qb.cluster(mbundle_streamlines)
+    centroids = Streamlines(clusters.centroids)
 
-def bundle_analysis(model_bundle_folder, bundle_folder, orig_bundle_folder,
-                    metric_folder, group, subject, no_disks=100,
-                    out_dir=''):
+    _, indx = cKDTree(centroids.data, 1,
+                      copy_data=True).query(target_bundle.data, k=1)
+
+    return indx
+
+
+def buan_bundle_profiles(model_bundle_folder, bundle_folder,
```

Comment:
This function still mixes a lot of file I/O with the actual computation. Is there any way we could separate these two things?

Reply:
I agree with @arokem  for this point. This function should not have any I/O.  All the reading parts should be done by the workflow or a separate function. You can have a function named `read_buan_folder` like you did for `save_buan_profiles`. The user will have to do these 2 steps: read the data, then call buan_bundle_profile function.

Reply:
moved `buan_bundle_profiles` function to `stats.py` workflow.